packages/entity-server/src/entity/middleware.ts

```diff
@@ -0,0 +1,101 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2021 Beyond Essential Systems Pty Ltd
+ */
+
+import { NextFunction, Response } from 'express';
+import { AccessPolicy } from '@tupaia/access-policy';
+import { PermissionsError } from '@tupaia/utils';
+import { EntityModel, EntityFields, FetchEntityRequest, FetchEntityResponse } from '../types';
+
+export const attachEntityToRequest = async (
+  req: FetchEntityRequest,
+  res: Response,
+  next: NextFunction,
+) => {
+  try {
+    const { entityCode, projectCode } = req.params;
+    const entity = await req.models.entity.findOne({
+      code: entityCode,
+    });
+
+    if (!entity || !(await userHasAccessToEntity(entity, projectCode, req.accessPolicy))) {
+      throw new PermissionsError(`No access to requested entity: ${entityCode}`);
+    }
+
+    req.context.entity = entity;
+
+    next();
+  } catch (error) {
+    next(error);
+  }
+};
+
+export const attachFormatterToResponse = async (
+  req: FetchEntityRequest,
+  res: FetchEntityResponse,
+  next: NextFunction,
+) => {
+  try {
+    res.context.formatEntityForResponse = mapEntitiesToFields(
+      extractFieldsFromQuery(req.query.fields),
+    );
+    next();
+  } catch (error) {
+    next(error);
+  }
+};
+
+const userHasAccessToEntity = async (
+  entity: EntityModel,
+  projectCode: string,
+  accessPolicy: AccessPolicy,
+) => {
+  // Assume user always has access to all world items.
+  if (entity.code === 'World') {
+    return true;
+  }
+  // Timor-Leste is temporarily turned off
+  if (entity.country_code === 'TL') {
+    return false;
+  }
```

Comment:
For sure, I'd prefer to remove both!

Reply:
`TL` actually may not be redundant, but certainly should be handled in Tupaia rather than here. Basically no one should even know that `TL` is in our system, no matter what their permissions, for political reasons, but we still wanted to hold on to all of the data for it. That's gone on for long enough that I'd prefer now to take a backup and get rid of the specific handling. Could you make a card for that please 🙏

Reply:
Created: https://github.com/beyondessential/tupaia-backlog/issues/2456